apps/openmw/mwrender/shadow.cpp

```diff
@@ -0,0 +1,121 @@
+#include "shadow.hpp"
+#include "parallelsplitshadowmap.hpp"
+#include <osgShadow/ShadowSettings>
+#include <osgShadow/ViewDependentShadowMap>
+
+#include <components/settings/settings.hpp>
+
+using namespace osgShadow;
+
+namespace MWRender
+{
+
+/*static */osg::ref_ptr<osg::Uniform> ShadowManager::_unshadowedAmbientBias=0;
+/*static */osg::Uniform* ShadowManager::getUnshadowedUniform()
+{
+    if(!_unshadowedAmbientBias.valid()) _unshadowedAmbientBias = new osg::Uniform("ambientBias",osg::Vec2(1,1));
```

Comment:
This is definitely insufficient to avoid undefined behaviour with RTTs without shadows - you're still sampling from unbound textures even if you don't do anything with the result. This *will* produce OpenGL errors on conformant OpenGL implementations.

Reply:
https://stackoverflow.com/questions/28411686/opengl-reading-from-unbound-texture-unit